File: browsing_history_manager.py
# Browsing History Stack

class BrowsingHistory:
    def __init__(self):
        self.history = []

    def add_page(self, page):
        self.history.append(page)

    def pop(self):
        if not self.is_empty():
            removed_page = self.history.pop()
            print(f'\n** Removed Page from History: \n   - {removed_page}')
            print(21 * '-')
        
    def peek_last_page(self):
        if not self.is_empty():
            peek_page = self.history[-1]
            print(f'\n* Last Page Visited: \n   - {peek_page}')
            print(21 * '-')
            return
        
    def is_empty(self):
        return len(self.history) == 0
    
    def size(self):
        return len(self.history)
    
    def display_history(self):
        if self.is_empty():
            print('\n ~ No browsing history to display.\n ')
        else:
            print('\n Browsing History: \n')
            for i, page in enumerate(self.history, start=1):
                print(f'{i}: {page}')
            print(21 * '-')


history = BrowsingHistory()

history.display_history()

history.add_page('www.google.com')
history.add_page('www.codingtemple.com')
history.add_page('www.netflix.com')
history.add_page('www.hulu.com')

empty = history.is_empty()
print(f'\nEmpty Browsing History? \n\t- {empty}')
print(21 * '-')

size = history.size()
print(f'\nPages in Browsing History: {size}')
print(21 * '-')

history.display_history()

history.peek_last_page()

history.pop()

history.display_history()

    

File: browsingHistoryManager.js
// Browsing History Stack

class BrowsingHistory {
    constructor() {
        this.history = []
        this.break = '-'
    }

    addPage(page) {
        this.history.push(page);
    }

    pop() {
        if (!this.isEmpty()) {
            const removedPage = this.history.pop();
            console.log(`\n** Removed Page from History: \n   - ${removedPage}`);
            console.log(this.break.repeat(21));
        }
    }

    peekLastPage() {
        if (!this.isEmpty()) {
            let peekPage = this.history[this.history.length - 1];
            console.log(`\n* Last Page Visited: \n   - ${peekPage}`);
            console.log(this.break.repeat(21));
            return
        }
    }

    isEmpty() {
        return this.history.length === 0;
    }

    size() {
        return this.history.length;
    }

    displayHistory() {
        if (this.isEmpty()) {
            console.log('\n ~ No browsing history to display.\n ')
        } else {
            console.log('\n Browsing Histroy: \n');
            this.history.forEach(page => {
                let i = 1;
                console.log(`${i}: ${page}`);
                i++;
            });
            console.log(this.break.repeat(21));
        }
    }
}

const history = new BrowsingHistory();

history.displayHistory();

history.addPage('www.google.com');
history.addPage('www.codingtemple.com');
history.addPage('www.netflix.com');
history.addPage('www.hulu.com');

const empty = history.isEmpty();
console.log(`\nEmpty Browsing History?\n\t- ${empty}`);
console.log(('-').repeat(21));

const size = history.size();
console.log(`\nPages in Browsing History: ${size}`);

history.displayHistory();

history.peekLastPage();

history.pop();

history.displayHistory();